File: C++/Queue.cpp
#include<iostream>
using namespace std;

const int MAX_SIZE = 10001;

typedef int element;
typedef struct {
    int front;
    int back;
    element array[MAX_SIZE];
} Queue;

void init(Queue *q)
{
    q->front = -1;
    q->back = -1;
}

void print(Queue *q)
{
    cout << "q: [ ";
    for(int i= q->front+1; i< q->back+1; i++)
        cout << q->array[i]<<" ";
    cout<<"]\n";
}

bool empty(Queue *q)
{
    // queue is empty
    if(q->front == q->back)
        return true;
    
    return false;
}

bool full(Queue *q)
{
    // queue is saturated
    if(q->back == MAX_SIZE -1)
        return true;
    
    return false;
}

void enqueue(Queue *q, element item)
{
    if(full(q))
    {
        cout <<"Queue is saturated \n";
        return;
    }
    
    q->back += 1;
    q->array[q->back] = item;
    print(q);
}

void deque(Queue *q)
{
    if (empty(q))
    {
        cout << "Queue is empty";
        return;
    }
    
    q->front += 1;
    int result = q->array[q->front];
    cout << result << " was removed\n";
    print(q);
    return;
}

int main()
{
    Queue q;

    enqueue(&q,1); // q: [ 1 ]
    enqueue(&q,2); // q: [ 1 2 ]
    enqueue(&q,5); // q: [ 1 2 5 ]
    dequeue(&q);    // q: [ 2 5 ]
}

File: Python/Queue.py
class Queue:
    def __init__(self):
        self.front = -1
        self.rear = -1
        self.queue = [None for _ in range(1000)]

    def empty(self):
        if self.front == self.rear:
            return True
        return False

    def print_queue(self):
        if self.empty():
            print("Queue is empty")
            return
        print("Queue = [",end=" ")
        for i in range(self.front + 1, self.rear+1):
            print(self.queue[i], end=" ")
        print("]")

    def enqueue(self, data):
        self.rear += 1
        self.queue[self.rear] = data

    def dequeue(self):
        self.front += 1


if __name__ == '__main__':
    q = Queue()
    q.enqueue(1)
    q.enqueue(3)
    q.enqueue(5)
    q.enqueue(7)
    q.enqueue(9)    # Queue = [ 1 3 5 7 9 ]
    q.print_queue()
    q.dequeue()     # Queue = [ 3 5 7 9 ]
    q.print_queue()